### ISSUE TITLE
Occasional extension update issues

### ISSUE THREAD

**Issue by grahamperrin** [BrowserWorks/waterfox#479]
Apologies for the vague title (we can change it to something more specific) … and @jbeich part of this issue might be better placed in FreeBSD Bugzilla … but for now I'll throw outline details into the MrAlex94/Waterfox/ repo because I'm aware of work towards Waterfox-oriented add-on services. 

## Example: [Containers Sync](https://addons.mozilla.org/addon/containers-sync/), 2018-02-25

![2018-02-25 10 26 42 update not found by about addons](https://user-images.githubusercontent.com/192271/37691461-2bbca2aa-2caa-11e8-9c74-2f4576e254bf.png)

tl;dr [version 1.0.1](https://addons.mozilla.org/addon/containers-sync/versions/?page=1#version-1.0.1) became available in November 2017 but three months later, the update (from 1.0) was not found by about:addons in Waterfox on FreeBSD-CURRENT. At AMO I see _some_ platform specificity for 1.0.1: 

> Works with Firefox for Android 55.0 and later, Firefox 55.0 and later 

– so I'm reminded of things such as `https://github.com/mozilla/addons-frontend/issues/2998` but that was a _front end_ AMO issue. 

## Example: [Ant Video Downloader](https://addons.mozilla.org/addon/video-downloader-player/)

tl;dr for https://www.reddit.com/r/waterfox/comments/85w44q/-/dw0xtjs/ I created a new profile named 'elevation' and set preferences to allow installation of [3.2.34](https://addons.mozilla.org/addon/video-downloader-player/versions/?page=1#version-3.2.34), 

> Works with Firefox for Android 57.0 and later, Firefox 57.0 and later 

– but refrained from installation. 

Instead I: 

1. began with preferences set to allow installation of legacy [2.4.7.50](https://addons.mozilla.org/addon/video-downloader-player/versions/?page=1#version-2.4.7.50)
2. installed 2.4.7.50
3. changed preferences to allow installation of 3.2.34

– about:addons check finds **no update** (from 2.4.7.50 to 3.2.34). Waterfox 56.0.4.36_3 (56.0.4.1) on FreeBSD-CURRENT. 

----

Maybe tomorrow night I'll make time to compare with Waterfox on a Mac …

**Comment by grahamperrin** [comment]
Containers Sync: issue not reproducible on a Mac. After installation of outdated 1.0, about:addons finds the update to 1.0.1. 

Reproducible with Waterfox on FreeBSD-CURRENT, so maybe it's an AMO back end bug.  

----

Ant Video Downloader: no automated update on the Mac. After manually updating, things made sense. My bad: I forgot that [checkCompatibility2](https://addons.mozilla.org/addon/checkcompatibility2/) simply disables compatibility checking (doesn't make the browser behave as if it's compatible).
